src/OpenTelemetry.Instrumentation.EntityFrameworkCore/Implementation/EntityFrameworkDiagnosticListener.cs

```diff
@@ -0,0 +1,208 @@
+// <copyright file="EntityFrameworkDiagnosticListener.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Data;
+using System.Diagnostics;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Instrumentation.EntityFrameworkCore.Implementation
+{
+    internal sealed class EntityFrameworkDiagnosticListener : ListenerHandler
+    {
+        internal const string DiagnosticSourceName = "Microsoft.EntityFrameworkCore";
+
+        internal const string ActivitySourceName = "OpenTelemetry.EntityFrameworkCore";
+        internal const string ActivityName = ActivitySourceName + ".Execute";
+
+        // TODO: get this value from payload
+        internal const string DatabaseSystemName = "mssql";
+
+        internal static readonly ActivitySource SqlClientActivitySource = new ActivitySource(ActivitySourceName, Version.ToString());
+
+        private static readonly Version Version = typeof(EntityFrameworkDiagnosticListener).Assembly.GetName().Version;
+
+        private readonly PropertyFetcher commandFetcher = new PropertyFetcher("Command");
+        private readonly PropertyFetcher connectionFetcher = new PropertyFetcher("Connection");
+        private readonly PropertyFetcher dataSourceFetcher = new PropertyFetcher("DataSource");
+        private readonly PropertyFetcher databaseFetcher = new PropertyFetcher("Database");
+        private readonly PropertyFetcher commandTypeFetcher = new PropertyFetcher("CommandType");
+        private readonly PropertyFetcher commandTextFetcher = new PropertyFetcher("CommandText");
+        private readonly PropertyFetcher exceptionFetcher = new PropertyFetcher("Exception");
+
+        private readonly EntityFrameworkInstrumentationOptions options;
+
+        public EntityFrameworkDiagnosticListener(string sourceName, EntityFrameworkInstrumentationOptions options)
+            : base(sourceName)
+        {
+            this.options = options;
```

Comment:
you could change this to 
```suggestion
            this.options = options ?? new EntityFrameworkInstrumentationOptions();
```
with that, you wouldn't need to add null checks everywhere for the `this.options`